File: core/chunk.py
import sys
import json

# initialize arguments
batch_size = int(sys.argv[1])
block_size = int(sys.argv[2])

# initialize loop variables
batch_x, batch_y = [], []
i = 0

# each line is one sample, we keep reading while data is streaming in
for line in sys.stdin:
    # convert to integer
    cur = [int(n) for n in line.split()]

    # slice into correct blocks
    batch_x.append(cur[:block_size])
    batch_y.append(cur[1:block_size+1])

    # iterate
    i += 1
    
    # if we have collected 'batch_size' many samples, we print to stdout
    if i % batch_size == 0:
        print(json.dumps({"batch_x": batch_x, "batch_y": batch_y}), flush = True)
        batch_x, batch_y = [], []

File: core/chunk.js
#!/usr/bin/env node

import { createReadStream } from 'fs';
import { createInterface } from 'readline';

// initialize arguments
const batchSize = parseInt(process.argv[2]);
const blockSize = parseInt(process.argv[3]);

// initialize loop variables
let batchX = [];
let batchY = [];

// create readline interface for stdin
const rl = createInterface({ input: process.stdin });

// main chunking loop
let i = 0;
for await (const line of rl) {
  // TODO: read in the stream of tokens and batch them in the right format
  // output tokens as a JSON string (look into JSON.stringify)
  // please label your batches "xb" and "yb" to pass the tests!
  
  // __start_solution__
  // convert to integer array and slice
  const cur = line.split(' ').map(n => parseInt(n));
  batchX.push(cur.slice(0, blockSize));
  batchY.push(cur.slice(1, blockSize + 1));
  i++;
  
  // if we have collected 'batch_size' many samples, print to stdout
  if (i % batchSize === 0) {
      console.log(JSON.stringify({ xb: batchX, yb: batchY }));
      batchX = []; batchY = [];
  }
  // __end_solution__
}
